source/base/mpi_compute_index_owner_internal.cc

```diff
@@ -146,20 +146,21 @@ namespace Utilities
               // smaller pieces in terms of the dictionary number.
               std::pair<types::global_dof_index, types::global_dof_index>
                 index_range(*interval->begin(), interval->last() + 1);
-              const unsigned int owner_last =
-                dof_to_dict_rank(interval->last());
-              unsigned int owner_first = numbers::invalid_unsigned_int;
-              while (owner_first != owner_last)
+
+              AssertThrow(index_range.second <= size, ExcInternalError());
+
+              while (index_range.first != index_range.second)
                 {
                   Assert(index_range.first < index_range.second,
                          ExcInternalError());
 
-                  owner_first = dof_to_dict_rank(index_range.first);
+                  const unsigned int owner =
+                    dof_to_dict_rank(index_range.first);
 
                   // this explicitly picks up the formula of
                   // dof_to_dict_rank, so the two places must be in sync
-                  types::global_dof_index next_index =
-                    std::min(get_index_offset(owner_first + 1),
+                  const types::global_dof_index next_index =
+                    std::min(get_index_offset(owner + stride_small_size),
```

Comment:
@kronbichler  I think this is also a bug. Do you agree.

Reply:
Is it? I think the change is fine, because we would hit processes without any unknowns in the previous case. But the code https://github.com/dealii/dealii/blob/6c065836a1ff9259393f6e851c60b20ec0cddb13/include/deal.II/base/mpi_compute_index_owner_internal.h#L413-L419 should make sure that we get the right result, shouldn't it? Did you find a case where this did not work?

Reply:
I removed this change. It is working, since we are rounding up :+1: